pkg/lwip/contrib/lwip.c

```diff
@@ -179,103 +163,146 @@ void lwip_bootstrap(void)
 {
     (void)_netif_add;   /* in case it is not used */
     /* TODO: do for every eligible netdev */
-#ifdef LWIP_NETIF_NUMOF
-#ifdef MODULE_NETDEV_TAP
-    for (unsigned i = 0; i < LWIP_NETIF_NUMOF; i++) {
-        netdev_tap_setup(&netdev_taps[i], &netdev_tap_params[i]);
-        if (_netif_add(&netif[i], &netdev_taps[i], lwip_netdev_init,
-                       tcpip_input) == NULL) {
-            DEBUG("Could not add netdev_tap device\n");
-            return;
+    if (LWIP_NETIF_NUMOF > 0) {
+        static struct netif netif[LWIP_NETIF_NUMOF];
```

Comment:
This does not work for the `lwip_sock` tests :-( (so for externally initialized `netif`s either). Maybe we should make this a global definition after all again and `#ifdef`/`#else` it as a `NULL` pointer if no netifs are defined?

Reply:
> This does not work for the `lwip_sock` tests :-( (so for externally initialized `netif`s either).

Do you mean the error message error:
```
ISO C forbids zero-size array ‘netif’ [-Werror=pedantic]
         static struct netif netif[LWIP_NETIF_NUMOF];
                             ^~~~~
```
This could be solved with `CFLAGS += -Wno-pedantic` in `pkg/lwip/contrib/Makefile` only for `lwip_contrib` module. With this C flag, `lwip_*` tests are working for me.

Reply:
Yes, but we don't want to deactivate this warning because of that ^^.

Reply:
Then maybe the `if` should be an `#if` after all.